### PR TITLE
Remove the head masking block in some vision models

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

#36545 introduced a mistake regarding the headmask

before that PR, we had some

```
        # Mask heads if we want to
        if head_mask is not None:
            attention_probs = attention_probs * head_mask
```
but after the PR, it became

```
    # Mask heads if we want to
    if attention_mask is not None:
        attn_weights = attn_weights * attention_mask
```
which is apparently wrong.

Furthermore, we already remove the support of head masking.

This PRs simply remove the blocks.

### CODE REVIEW

**Review by vasqu — APPROVED** [review]
Thx, what nasty bug :sweat:
